src/server/db/resources/user/user.model.js

```diff
@@ -14,13 +15,6 @@ const userSchema = new mongoose.Schema(
       type: String,
       required: true,
     },
-    posts: [
```

Comment:
Posts need to be fetched with `Post.find({ author: user._id })`. They cannot be on the user directly without having duplicate data, because virtual fields cannot perform queries since they only "live" in Mongoose, not in MongoDB.